backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportNamer.kt

```diff
@@ -161,6 +168,12 @@ private class ObjCExportNamingHelper(
         "Type" -> "${name}_" // See https://github.com/JetBrains/kotlin-native/issues/3167
         else -> name
     }
+
+    fun isTypeParameterNameReserved(name: String): Boolean = name in reservedTypeParameterNames
+
+    private val reservedTypeParameterNames = setOf("id", "NSObject", "NSArray", "NSCopying", "NSNumber", "NSInteger",
```

Comment:
Where this list comes from?

Reply:
Generally from here: https://github.com/JetBrains/kotlin-native/pull/3778/files#diff-d13ae8cff625fe7530e3a86b4102ca0fL666

I have some plans to improve this part of code, but it requires more deep changes.